### PR TITLE
Allow loss function customisation

### PR DESCRIPTION
**Pull request by jthorton**
## Description
This PR adds an extra loss function argument to the `DGLMoleculeLightningModel` allowing users to provide their own loss function for extra customisation rather than having to implement them all in nagl. 

## Todos
Notable points that this PR has either accomplished or will accomplish.
  - [ ] 

## Questions
- [ ] How would we also include a way to weight the loss function when fitting to properties on different scales. One idea is to add a weights argument to the init which takes a dictionary of the readout label and the corresponding weight factor, however, this means that users would have to enter the labels twice. Ideally they would enter the label the readout and weight in one go somehow. 

## Status
- [x] Ready to go

### CODE REVIEW

File: nagl/lightning.py [hunk 1]
```diff
@@ -29,12 +31,14 @@ def __init__(
...
         readout_modules: Dict[str, ReadoutModule],
         learning_rate: float,
+        loss_function: LossFunction = rmse_loss,
     ):
 
         super().__init__()
 
         self.convolution_module = convolution_module
         self.readout_modules = torch.nn.ModuleDict(readout_modules)
+        self.loss_function = loss_function
```
Comment (SimonBoothroyd) [inline nagl/lightning.py:41]:
```suggestion
        self._loss_function = loss_function
```

File: nagl/lightning.py [hunk 2]
```diff
@@ -63,7 +67,7 @@ def _default_step(
         loss = torch.zeros(1).type_as(next(iter(y_pred.values())))
 
         for label_name, label in labels.items():
-            loss += _rmse_loss(y_pred[label_name], label)
+            loss += self.loss_function(y_pred[label_name], label)
```
Comment (SimonBoothroyd) [inline nagl/lightning.py:70]:
```suggestion
            loss += self._loss_function(y_pred[label_name], label)
```

**Review by SimonBoothroyd — APPROVED** [review]
Looks great - thanks for adding this!

**Comment by jthorton** [comment]
Need to decide if we are updating for the new toolkit in #44 or if we should pin to the 0.10-x series of the toolkit.

**Comment by jthorton** [comment]
A lot of the current test fits are done using the 10-x toolkit series so maybe it would be best to just pin the toolkit for now.
